package.json

```diff
@@ -72,236 +72,236 @@
-    "trunc-text": "1.0.2",
-    "uglifyjs-webpack-plugin": "0.4.6",
-    "ui-select": "0.19.6",
-    "url-loader": "0.5.9",
-    "uuid": "3.0.1",
-    "validate-npm-package-name": "2.2.2",
-    "vision": "4.1.0",
-    "webpack": "3.6.0",
-    "webpack-merge": "4.1.0",
-    "wreck": "12.4.0",
-    "yauzl": "2.7.0"
+    "@elastic/datemath": "^2.3.0",
+    "@elastic/filesaver": "^1.1.2",
+    "@elastic/numeral": "^2.3.1",
+    "@elastic/test-subj-selector": "^0.2.1",
+    "@elastic/ui-ace": "^0.2.3",
+    "JSONStream": "^1.1.1",
+    "accept-language-parser": "^1.2.0",
+    "angular": "^1.6.5",
+    "angular-aria": "^1.6.6",
+    "angular-bootstrap-colorpicker": "^3.0.19",
+    "angular-elastic": "^2.5.0",
+    "angular-route": "^1.4.7",
+    "angular-sanitize": "^1.5.7",
+    "angular-sortable-view": "^0.0.15",
+    "angular-translate": "^2.13.1",
+    "ansicolors": "^0.3.2",
+    "autoprefixer": "^6.5.4",
+    "babel-core": "^6.21.0",
+    "babel-jest": "^21.0.0",
+    "babel-loader": "^7.1.2",
+    "babel-plugin-add-module-exports": "^0.2.1",
+    "babel-plugin-transform-async-generator-functions": "^6.24.1",
+    "babel-plugin-transform-class-properties": "^6.24.1",
+    "babel-plugin-transform-define": "^1.3.0",
+    "babel-plugin-transform-object-rest-spread": "^6.23.0",
+    "babel-polyfill": "^6.20.0",
+    "babel-preset-env": "^1.4.0",
+    "babel-preset-react": "^6.22.0",
+    "babel-register": "^6.18.0",
+    "bluebird": "^2.9.34",
+    "body-parser": "^1.12.0",
+    "boom": "^5.2.0",
+    "brace": "^0.10.0",
+    "bunyan": "^1.7.1",
+    "cache-loader": "^1.0.3",
+    "check-hash": "^1.0.1",
+    "color": "^1.0.3",
+    "commander": "^2.8.1",
+    "css-loader": "^0.28.7",
+    "d3": "^3.5.6",
+    "d3-cloud": "^1.2.1",
+    "dragula": "^3.7.0",
+    "elasticsearch": "^13.0.1",
+    "elasticsearch-browser": "^13.0.1",
+    "encode-uri-query": "^1.0.0",
+    "even-better": "^7.0.2",
+    "expiry-js": "^0.1.7",
+    "exports-loader": "^0.6.4",
+    "expose-loader": "^0.7.3",
+    "extract-text-webpack-plugin": "^3.0.1",
+    "file-loader": "^1.1.4",
+    "focus-trap-react": "^3.0.3",
+    "font-awesome": "^4.4.0",
+    "glob": "^5.0.13",
+    "glob-all": "^3.0.1",
+    "good-squeeze": "^2.1.0",
+    "h2o2": "^5.1.1",
+    "handlebars": "^4.0.5",
+    "hapi": "^14.2.0",
+    "http-proxy-agent": "^1.0.0",
+    "imports-loader": "^0.7.1",
+    "inert": "^4.0.2",
+    "jade": "^1.11.0",
+    "jade-loader": "^0.8.0",
+    "joi": "^10.4.1",
+    "jquery": "^2.2.4",
+    "js-yaml": "^3.4.1",
+    "json-stringify-safe": "^5.0.1",
+    "jstimezonedetect": "^1.0.5",
+    "leaflet": "^1.0.3",
+    "leaflet-draw": "^0.4.10",
+    "leaflet-responsive-popup": "^0.2.0",
+    "leaflet.heat": "^0.2.0",
+    "less": "^2.7.1",
+    "less-loader": "^4.0.5",
+    "lodash": "^3.10.1",
+    "lru-cache": "^4.1.1",
+    "markdown-it": "^8.3.2",
+    "mathjs": "^3.16.2",
+    "minimatch": "^2.0.10",
+    "mkdirp": "^0.5.1",
+    "moment": "^2.13.0",
```

Comment:
By loosening out required versions (in package.json, yarn.lock makes them strict) we allow yarn to merge our dependencies with the dependencies of our dependencies. This is more efficient, but also necessary for some of our dependencies

With strict versions in Kibana yarn was installing 12 copies of `moment`.

```
node_modules/moment-timezone/node_modules/moment
node_modules/statehood/node_modules/moment
node_modules/moment
node_modules/hapi/node_modules/moment
node_modules/libesvm/node_modules/moment
node_modules/grunt-esvm/node_modules/moment
node_modules/inert/node_modules/moment
node_modules/bunyan/node_modules/moment
node_modules/vision/node_modules/moment
node_modules/h2o2/node_modules/moment
node_modules/even-better/node_modules/moment
node_modules/makelogs/node_modules/moment
```

This caused bugs and test failures because we rely on moment having the same internal state everywhere, especially in `moment-timezone`, but `moment-timezone` depends on `moment@^2.6.0`. With npm this a satisfied by our dependency on `moment@2.13.0`, but yarn attempts to get the latest version of moment it can for `moment-timezone`, causing it to install `moment@2.19.3` into `moment-timezone/node_modules`. When Kibana and moment-timezone depend on a `^` version of moment they both get the most recent version, installed once in the root node_modules directory.

With looser versions, yarn installs two versions of moment
```
node_modules/@elastic/datemath/node_modules/moment # @elastic/datemath needs a looser version check
node_modules/moment
```

Overall, with strict versions kibana installs *2521* modules and with looser versions kibana installs *2297* modules. A decent win.

Reply:
Merging https://github.com/elastic/kibana/pull/15487 takes us to one version of moment